src/main/kotlin/com/terraformation/backend/i18n/TimeZones.kt

```diff
@@ -0,0 +1,503 @@
+              "Indian/Mahe",
+              "Indian/Maldives",
+              "Indian/Mauritius",
+              "Indian/Mayotte",
+              "Indian/Reunion",
+              "Pacific/Apia",
+              "Pacific/Auckland",
+              "Pacific/Bougainville",
+              "Pacific/Chatham",
+              "Pacific/Easter",
+              "Pacific/Efate",
+              "Pacific/Enderbury",
+              "Pacific/Fakaofo",
+              "Pacific/Fiji",
+              "Pacific/Funafuti",
+              "Pacific/Galapagos",
+              "Pacific/Gambier",
+              "Pacific/Guadalcanal",
+              "Pacific/Guam",
+              "Pacific/Honolulu",
+              "Pacific/Johnston",
+              "Pacific/Kiritimati",
+              "Pacific/Kosrae",
+              "Pacific/Kwajalein",
+              "Pacific/Majuro",
+              "Pacific/Marquesas",
+              "Pacific/Midway",
+              "Pacific/Nauru",
+              "Pacific/Niue",
+              "Pacific/Norfolk",
+              "Pacific/Noumea",
+              "Pacific/Pago_Pago",
+              "Pacific/Palau",
+              "Pacific/Pitcairn",
+              "Pacific/Ponape",
+              "Pacific/Port_Moresby",
+              "Pacific/Rarotonga",
+              "Pacific/Saipan",
+              "Pacific/Tahiti",
+              "Pacific/Tarawa",
+              "Pacific/Tongatapu",
+              "Pacific/Truk",
+              "Pacific/Wake",
+              "Pacific/Wallis",
+              "Etc/UTC",
+          )
+          .map { ZoneId.of(it) }
+
+  private fun getTimeZoneNameWithoutCity(zoneId: ZoneId, locale: Locale): String {
+    val cldr = CLDR.get(locale)
+    val zoneInfo = cldr.Calendars.timeZoneInfo(zoneId.id)
+
+    return if (zoneId.rules.nextTransition(Instant.now()) == null) {
+      zoneInfo.names().long_().standard()
+    } else {
+      zoneInfo.names().long_().generic()
+    }
+  }
+
+  private val timeZonesByEnglishNameWithoutCity: Map<String, List<ZoneId>> by lazy {
+    chromeTimeZones.groupBy { getTimeZoneNameWithoutCity(it, Locale.ENGLISH) }
+  }
+
+  private fun getTimeZoneNameWithCity(zoneId: ZoneId, locale: Locale): String {
+    if (zoneId.id == "Etc/UTC") {
+      return zoneId.getDisplayName(TextStyle.FULL, locale)
+    }
+
+    val nameWithoutCity = getTimeZoneNameWithoutCity(zoneId, locale)
+    val englishWithoutCity = getTimeZoneNameWithoutCity(zoneId, Locale.ENGLISH)
+    val calendars = CLDR.get(locale).Calendars
+    val city = calendars.timeZoneInfo(zoneId.id).city().name()
+    val citiesForZonesOfSameName =
+        timeZonesByEnglishNameWithoutCity[englishWithoutCity]
+            ?.map { calendars.timeZoneInfo(it.id).city().name() }
+            ?.distinct()
+            ?: emptyList()
+    return if (citiesForZonesOfSameName.size != 1) {
+      "$nameWithoutCity - $city"
+    } else {
+      nameWithoutCity
+    }
+  }
+
+  private val timeZonesWithoutDuplicateCities: List<ZoneId> by lazy {
```

Comment:
is this the cldr solution rewritten for us?

Reply:
I'd say it's more inspired by, or maybe informed by, the CLDR solution. They made different design tradeoffs but their code helped me think through some scenarios.